### PR TITLE
MM-24192 Adding Teleport utility group

### PR DESCRIPTION
**Pull request by stylianosrigas**
#### Summary
This PR adds:
- Teleport Utility Group 
- Function to get Account name Aliases

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-24192

#### Release Note
```release-note
Adds teleport utility group
```

### CODE REVIEW

File: internal/tools/aws/iam.go [hunk 1]
```diff
@@ -235,3 +235,12 @@ func (a *Client) iamEnsureAccessKeyCreated(awsID string, logger log.FieldLogger)
 
 	return createResult.AccessKey, nil
 }
+
+// GetAccountAliases returns the AWS account name aliases.
+func (a *Client) GetAccountAliases() (*iam.ListAccountAliasesOutput, error) {
+	accountAliases, err := a.Service().iam.ListAccountAliases(&iam.ListAccountAliasesInput{})
+	if err != nil {
+		return nil, err
```
Comment (gabrieljackson) [inline internal/tools/aws/iam.go:243]: nit: wrapping this error would be nice

File: internal/provisioner/utility_group.go [hunk 2]
```diff
@@ -48,8 +48,9 @@ type utilityGroup struct {
 
 // List of repos to add during helm setup
 var helmRepos = map[string]string{
-	"jetstack": "https://charts.jetstack.io",
-	"stable":   "https://kubernetes-charts.storage.googleapis.com",
+	"jetstack":    "https://charts.jetstack.io",
```
[the remark this answers is not in the document]
Comment (stylianosrigas) [reply internal/provisioner/utility_group.go]: Yeap, not needed anymore, removed.

File: internal/provisioner/teleport.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	accountAliases, err := awsClient.GetAccountAliases()
+	if err != nil {
+		return "", err
+	}
+	if len(accountAliases.AccountAliases) < 1 {
+		return "", errors.New("Account Alias not defined")
+	}
+	for _, alias := range accountAliases.AccountAliases {
+		if strings.HasPrefix(*alias, "mattermost-cloud") {
+			return strings.Split(*alias, "-")[2], nil
```
[the remark this answers is not in the document]
Comment (stylianosrigas) [reply internal/provisioner/teleport.go:118]: Thanks, nice catch Gabe. Fixed ;)

File: internal/tools/aws/iam.go (hunk 1 above)
Comment (stylianosrigas) [reply internal/tools/aws/iam.go]: Fixed thanks :)

**Pull request merged**